src/robot/libraries/Collections.py

```diff
@@ -737,19 +737,28 @@ def dictionary_should_not_contain_value(self, dictionary, value, msg=None):
         default = "Dictionary contains value '%s'." % (value,)
         _verify_condition(not value in dictionary.values(), default, msg)
 
-    def dictionaries_should_be_equal(self, dict1, dict2, msg=None, values=True):
+    def dictionaries_should_be_equal(self, dict1, dict2, msg=None, values=True, ignore_keys=None):
         """Fails if the given dictionaries are not equal.
 
         First the equality of dictionaries' keys is checked and after that all
         the key value pairs. If there are differences between the values, those
         are listed in the error message. The types of the dictionaries do not
         need to be same.
 
+        ``ignore_keys`` can be used to provide a list of keys to ignore in the
+        comparison.
+
         See `Lists Should Be Equal` for more information about configuring
         the error message with ``msg`` and ``values`` arguments.
         """
+
+        if ignore_keys is None:
+            ignore_keys = []
         self._validate_dictionary(dict1)
         self._validate_dictionary(dict2, 2)
+        dict1 = {k: v for k, v in dict1.items() if k not in ignore_keys}
+        dict2 = {k: v for k, v in dict2.items() if k not in ignore_keys}
```

Comment:
This should be done only when `ignore_keys` is set. Now dictionaries are always recreated which wastes resources and in with custom dict subsclasses can also change semantics.